Makefile

```diff
@@ -0,0 +1,113 @@
+# Package configuration
+PROJECT := gitbase-playground
+COMMANDS := cmd/gitbase-playground
+DOCKER_ORG := src-d
+DEPENDENCIES := \
+	github.com/golang/dep/cmd/dep \
+	github.com/jteeuwen/go-bindata \
+	github.com/golang/lint/golint
+GO_LINTABLE_PACKAGES := $(shell go list ./... | grep -v '/vendor/')
+GO_BUILD_ENV := CGO_ENABLED=0
+
+# Tools
+GODEP := dep
+GOLINT := golint
+GOVET := go vet
+BINDATA := go-bindata
+DIFF := diff
+GITADD := git add -A
+
+# Default rule
+all:
+
+# Including ci Makefile
+CI_REPOSITORY ?= https://github.com/dpordomingo/ci.git
+CI_BRANCH ?= v1
+CI_PATH ?= $(shell pwd)/.ci
+MAKEFILE := $(CI_PATH)/Makefile.main
+$(MAKEFILE):
+	@git clone --quiet --depth 1 -b $(CI_BRANCH) $(CI_REPOSITORY) $(CI_PATH);
+-include $(MAKEFILE)
+
+# Makefile.main::dependencies -> Makefile.main::$(DEPENDENCIES) -> this::dependencies
+dependencies: | front-dependencies back-dependencies exit
+
+# Makefile.main::test -> this::test
+test: front-test
+
+# this::build -> Makefile.main::build -> Makefile.main::$(COMMANDS)
+build: front-build back-build
+	@echo
+
+coverage: | test-coverage codecov
+
+lint: back-lint front-lint
+
+validate-commit: | \
+	back-dependencies \
+	back-ensure-assets-proxy \
+	front-fix-lint-errors \
+	no-changes-in-commit
+
+exit:
+	exit 0;
+
+
+## Compiles the assets, and serve the tool through its API
+
+serve: | front-build back-start
+
+
+# Backend
+
+assets := ./server/assets/asset.go
+assets_back := $(assets).bak
+
+back-dependencies:
+	$(GODEP) ensure
+
+back-build: back-bindata
+
+back-bindata:
+	$(BINDATA) \
+		-pkg assets \
+		-o $(assets) \
+		build/public/*
+
+back-lint: $(GO_LINTABLE_PACKAGES)
+$(GO_LINTABLE_PACKAGES):
+	$(GOLINT) $@
+	$(GOVET) $@
+
+back-start:
+	GITBASEPG_ENV=dev go run cmd/gitbase-playground/main.go
```

Comment:
I thought we agreed to be `prod` by default for our applications, no?

Reply:
You're right, but `back-start` target is meant for dev purposes only.
Would you prefer it to be launched with `prod` conf?

Reply:
If it's only for development I would prefer to run it with proper `dev` environment.

But looks like it's not only dev command. In `serve` rule you call `back-start`. And `make serve` is a recommended way to run our applications (at least 2 others of them) if the one doesn't want to use docker.

If I'm not mistaken in code annotation you suggested to run apps in prod mode for such case.

Reply:
Yes, you're right @smacker : I purposed (and purpose) to set the defaults to `prod` settings when it is being used for final users. The rationale in behind would be
>_final users should be able to run the app with as less configuration effort as possible_

Since this tool was purposed to be a final user app, the `README.md` only [recommends to run the app with docker or through a released binary](https://github.com/dpordomingo/gitbase-playground/blob/a031d9eea95525ce72c12441eeef3d0f6082074c/README.md#run-the-playground)

Because of that reason: "_to make the tool easy to run for the final user_", with as less dependencies as possible (only the really needed), the `make serve` recommendation was moved to the [CONTRIBUTING.md#development](https://github.com/src-d/gitbase-playground/pull/15/files#diff-8595297407c24b9ccb87b2a698cca9c2R70) section.

If all of you think that we should recommend to run the tool that other way (with `go`, `GOPATH`, `yarn`...) then I'd move it to the `README.md` and remove the `dev` conf in the rule.

Reply:
I like using make only for dev 👍